app/repositories/v2/ListRepository.scala

```diff
@@ -97,6 +95,16 @@ class ListRepository @Inject() (
   def getListByNameWithFilter(listName: ListName, versionId: VersionId, filter: FilterParams): Source[JsObject, NotUsed] =
     getListByName(listName, versionId, Some(filter))
 
+  def deleteOldImports(payload: ReferenceDataPayload, createdOn: Instant): Future[ListRepositoryDeleteResult] =
+    collection
+      .deleteMany(Filters.lt("createdOn", createdOn))
+      .toFuture()
+      .map(_.wasAcknowledged())
+      .map {
+        case true  => SuccessfulDelete
+        case false => FailedDelete(payload.listNames)
+      }
+
```

Comment:
We can maybe convert this to be an Either, it may help with error logging later on. You could have a super type of error and have different types like FailedToDeleteListRepository and FailedToDeleteVersionRepository. May help with the matcher later on :D 

Something similar I did in the old backend: https://github.com/hmrc/transits-movements-trader-at-departure/blob/main/app/models/ErrorState.scala